data-pipeline-flink/user-cache-updater-2.0/src/main/scala/org/sunbird/dp/usercache/functions/UserCacheUpdaterFunctionV2.scala

```diff
@@ -43,25 +43,35 @@ class UserCacheUpdaterFunctionV2(config: UserCacheUpdaterConfigV2)(implicit val
 
   override def processElement(event: Event, context: ProcessFunction[Event, Event]#Context, metrics: Metrics): Unit = {
     metrics.incCounter(config.totalEventsCount)
-    Option(event.getId).map(id => {
-      Option(event.getState).map(name => {
-        val userData: mutable.Map[String, AnyRef] = name.toUpperCase match {
-          case "CREATE" | "CREATED" | "UPDATE" | "UPDATED" => UserMetadataUpdater.execute(id, event, metrics, config, dataCache, cassandraConnect, custodianOrgId)
-          case _ => {
-            logger.info(s"Invalid event state name either it should be(Create/Created/Update/Updated) but found $name for ${event.mid()}")
+    try {
+      Option(event.getId).map(id => {
+        Option(event.getState).map(name => {
+          val userData: mutable.Map[String, AnyRef] = name.toUpperCase match {
+            case "CREATE" | "CREATED" | "UPDATE" | "UPDATED" => {
+              logger.info(s"Processing event for user: ${id} having mid: ${event.mid()}")
```

Comment:
Move this statement to the exception block

Reply:
Done